src/natcap/invest/globio.py

```diff
@@ -939,22 +939,29 @@ def _collapse_infrastructure_layers(
 
     def _collapse_infrastructure_op(*infrastructure_array_list):
         """For each pixel, create mask 1 if all valid, else set to nodata."""
-        nodata_mask = (
-            numpy.isclose(
-                infrastructure_array_list[0], infrastructure_nodata_list[0]))
-        infrastructure_result = infrastructure_array_list[0] > 0
+        infrastructure_result = numpy.empty_like(
+            infrastructure_array_list[0], dtype=numpy.int32)
+        # Zero's indicate where there's no infrastructure influence but not
+        # necessarily nodata
+        infrastructure_result[:] = 0
```

Comment:
If you want to skip this two step, you can say `nodata.zeros(infrastructure_array_list[0].shape...` rather than make an empty array and then set it all to 0. Also I see you changed the type to an `int32` but I can't tell why. Can you talk a little more about that? 

Also typo on "Zero's"

Reply:
Now using numpy.zeros and dropped the int32 for uint8 here: https://github.com/natcap/invest/pull/186/commits/86dab80e89b22c452e5da5b2efec78decbfa62cb